src/Lib2.hs

```diff
@@ -16,6 +16,12 @@ type Database = [(TableName, DataFrame)]
 
 -- Keep the type, modify constructors
 data ParsedStatement = ParsedStatement
```

Comment:
Maybe something like this could work

```suggestion
data RelationalOperator = EQ | NE | LT | GT | LE | GE

data LogicalOperator = And

data ColumnName = String

data WhereCriteria = WhereCriteria ColumnName RelationalOperator Value

data AggregateFunction = Min | Sum

data Aggregate = Aggregate AggregateFunction ColumnName

data SelectData = SelectColumn ColumnName | SelectAggregate Aggregate

data SelectQuery = SelectQuery [SelectData]

data WhereClause = WhereClause [(WhereCriteria, Maybe LogicalOperator)]

data ParsedStatement = SelectStatement {
    table :: TableName,
    selectData :: SelectQuery,
    whereClause :: WhereClause
} | ShowTablesStatement {
    tables :: [TableName]
}
```

Reply:
Just question about `data WhereCriterion = WhereCriterion ColumnName RelationalOperator Value`, maybe change to Either (Value, ColumnName) because two column values can be compared, not only column value with some constant?

Reply:
True, but I think Either holds an error on left side and result in right value by convention: https://hackage.haskell.org/package/base-4.19.0.0/docs/Data-Either.html

Maybe we should create another data type? E.g. data Expression = ColumnName | Value?